### DISCUSSION TITLE
Penggunaan LabRole didalam contoh HelloWorld sam-cli

### DISCUSSION THREAD

**Discussion by slawalata** [awsugid/forum#24] [Q&A] [answered]
Salam.

Saya ingin mencoba deploy `HelloWorld` menggunakan  `SAM-CLI` dan `learner lab account` berdasarkan tautan ini:
https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-getting-started-hello-world.html#serverless-getting-started-hello-world-deploy

`Learner lab` adalah lab yang diberikan bagi para pelajar yang ikut kelas _tutorial-aws_.

Waktu menjalankan tutorial `sam-cli hello-world`, saya mendapatkan error kalau account saya tidak bisa membuat _IAMRole_ yang berusaha dibuat dalam file `cloud formation` _template.yml_ .

Pertanyaan:
- Bagaimana merubah cloud-formation yang ada didalam contoh untuk menggunakan LabRole dalam `leaner lab` yang ada?
- Atau bagaimana saya bisa menjalankan tutorial ini menggunakan `aws learner lab`?

Terima kasih atas bantuan teman teman.

**Comment by anggras** [comment, marked as answer]
Kalau LabRole tidak punya permission untuk membuat IAMRole, sepertinya akan tidak berhasil juga.
Tapi bisa dicoba untuk memakai `--role-arn` option dan menambah arn dari LabRole saat menjalankan `sam deploy`.

**Reply by slawalata** [reply]
Thanks, ternyata dalam cloud formation disediakan _role_ yang dapat digunakan selain membuat role baru untuk mejalankan _lamda function_.

Opsi lain menggunala `--role-arn`  juga menarik. Terima kasih atas tanggapannya.

salam
slawalata
